conduit/pipeline/pipeline_test.go

```diff
@@ -866,3 +867,114 @@ func TestRoundOverwrite(t *testing.T) {
+	SampleConfig: "",
+}
+
+// New initializes an importer
+func New() importers.Importer {
+	return &errorImporter{}
+}
+
+func (e errorImporter) Metadata() conduit.Metadata {
+	return errorImporterMetadata
+}
+
+func (e errorImporter) Init(_ context.Context, _ plugins.PluginConfig, _ *log.Logger) (*bookkeeping.Genesis, error) {
+	return e.genesis, nil
+}
+
+func (e errorImporter) Config() string {
+	return ""
+}
+
+func (e errorImporter) Close() error {
+	return nil
+}
+
+func (e errorImporter) GetBlock(_ uint64) (data.BlockData, error) {
+	return data.BlockData{}, fmt.Errorf("error maker")
+}
+
+// TestPipelineRetryVariables tests that modifying the retry variables results in longer time taken for a pipeline to run
+func TestPipelineRetryVariables(t *testing.T) {
+	tests := []struct {
+		name             string
+		retryDelayString string
+		retryCount       uint64
+		totalDuration    time.Duration
+		epsilon          time.Duration
+	}{
+		{"2 seconds", "2s", 1, 2 * time.Second, 1 * time.Second},
+		{"10 seconds", "2s", 5, 10 * time.Second, 1 * time.Second},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+
+			var pImporter importers.Importer = &errorImporter{genesis: &bookkeeping.Genesis{Network: "test"}}
+			var pProcessor processors.Processor = &mockProcessor{}
+			var pExporter exporters.Exporter = &mockExporter{}
+			l, _ := test.NewNullLogger()
+			retryDelay, err := time.ParseDuration(testCase.retryDelayString)
+			assert.NoError(t, err)
+			pImpl := pipelineImpl{
+				ctx: context.Background(),
+				cfg: &Config{
+					RetryCount:       testCase.retryCount,
+					RetryDelayString: testCase.retryDelayString,
+					RetryDelay:       retryDelay,
+					ConduitArgs: &conduit.Args{
+						ConduitDataDir:    t.TempDir(),
+						NextRoundOverride: 0,
+					},
+					Importer: NameConfigPair{
+						Name:   "",
+						Config: map[string]interface{}{},
+					},
+					Processors: []NameConfigPair{
+						{
+							Name:   "",
+							Config: map[string]interface{}{},
+						},
+					},
+					Exporter: NameConfigPair{
+						Name:   "unknown",
+						Config: map[string]interface{}{},
+					},
+				},
+				logger:       l,
+				initProvider: nil,
+				importer:     &pImporter,
+				processors:   []*processors.Processor{&pProcessor},
+				exporter:     &pExporter,
+				pipelineMetadata: state{
+					GenesisHash: "",
+					Network:     "",
+					NextRound:   3,
+				},
+				wg: sync.WaitGroup{},
+			}
+
+			// pipeline should initialize if NextRoundOverride is not set
+			err = pImpl.Init()
+			assert.Nil(t, err)
+			before := time.Now()
+			pImpl.Start()
+			pImpl.wg.Wait()
+			timeTaken := time.Since(before)
+
+			msg := fmt.Sprintf("seconds taken: %s, expected duration seconds: %s, epsilon: %s", timeTaken.String(), testCase.totalDuration.String(), testCase.epsilon.String())
+			assert.Truef(t, timeTaken <= testCase.totalDuration+testCase.epsilon || timeTaken >= testCase.totalDuration-testCase.epsilon, msg)
```

Comment:
I believe you can use [WithinDuration](https://pkg.go.dev/github.com/stretchr/testify@v1.8.1/require#WithinDuration) for this check.